packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -655,6 +668,16 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         this.autoscaledPoolOptions = { ...autoscaledPoolOptions, ...basicCrawlerAutoscaledPoolConfiguration };
     }
 
+    /**
+     * Checks if the given error is a proxy error by comparing its message to a list of known proxy error messages.
+     * Used for retrying requests that failed due to proxy errors.
+     * @param {Error} error - The error to check.
+     * @returns {boolean} - `true` if the error is a proxy error, `false` otherwise.
```

Comment:
i think this should be without the dashes, no? also no need to repeat the types (and i would rather put the return type to the method signature instead of jsdoc)